src/main/startup/startup_stm32f40xx.s

```diff
@@ -80,21 +83,13 @@ Reset_Handler:
   dsb
 
   // Check for bootloader reboot
-  ldr r0, =0x2001FFFC         // mj666
+  ldr r0, =bootloader_request
```

Comment:
Isn't it easier to call C function  (`checkForBootLoaderRequest()`) here?

Reply:
Yep, I think so, my comment about not being sure whether both C and Assembly code for reboot to bootloader were about this lines :-)
I'll try it now

Reply:
ARM ABI is great for mixing C and assembly. Even data/bss initialization can be done in C, just taking a little care to avoid initialized data and zeroed `static`

Reply:
This PR turned out to be a jewel for learning some useful ARM details, thank you Petr!